### PR TITLE
Support deltas for system images (bsc#1201498)

### PR DESCRIPTION
**Pull request by nadvornik**
This adds support for deltas on system (kis) images.

### CODE REVIEW

File: python-susemanager-retail/retail_create_delta [hunk 1]
```diff
@@ -182,12 +199,31 @@ try:
...
-    target_url = target_pillar.get('sync', {}).get('bundle_url')
     img_path = "/srv/www/os-images/{0}/".format(org_id)
 
-    delta_file, delta_url = generate_delta(src_image['files'][0]['file'], target_image['files'][0]['file'], src_pillar, target_pillar, img_path, args)
+    delta_file = "{0}-{1}-{2}-{3}-{4}-{5}.delta".format(
+                 target_image.get('name'), target_image.get('version'), target_image.get('revision'),
+                 src_image.get('name'), src_image.get('version'), src_image.get('revision'))
+
+    fqdn = socket.getfqdn()
+    delta_url = "https://{0}/os-images/{1}/{2}".format(fqdn, org_id, delta_file)
```
[the remark this answers is not in the document]
Comment (nadvornik) [reply python-susemanager-retail/retail_create_delta:209]: Currently, the delta file is copied locally. File upload needs a switch to http api.
URL is generated the same way as in java, so I think it should work for now. For fully correct handling it needs a new API call, I think.
These are reasons why I want to postpone this for next PR.

File: python-susemanager-retail/retail_create_delta [hunk 2]
```diff
@@ -16,39 +16,57 @@ import susemanager.retail.log as retail_log
...
-    'tgz': 'zcat',
-    'tar.xz': 'xzcat'
+    '.gz': '| zcat',
+    '.tgz': '| zcat',
+    '.xz': '| xzcat',
+    '': ''
 }
 
 def bundle_path(url):
     return re.sub(r"^.*/os-images/", "/srv/www/os-images/", url)
```
[the remark this answers is not in the document]
Comment (nadvornik) [reply python-susemanager-retail/retail_create_delta]: fixed

**Comment by nadvornik** [comment]
I have added a fix for deleting of unused boot images

**Review by aaannz — APPROVED** [review]
👍🏾

**Pull request merged**